ksqldb-common/src/main/java/io/confluent/ksql/function/UdfIndex.java

```diff
@@ -168,6 +197,37 @@ T getFunction(final List<SqlType> arguments) {
         .orElseThrow(() -> createNoMatchingFunctionException(arguments));
   }
 
+  private void getUdfCandidates(
+      final List<SqlArgument> arguments,
```

Comment:
Similarly to above, kept a `SqlArgument` track and a `SqlType` track because of other function types that still use `SqlType`. It's an option to transition all functions to passing in `SqlArgument` but would be a large refactor

Reply:
Can we just change UdfIndex so that
```
  T getFunction(final List<SqlType> arguments) {
  // code to transform the list of SqlType to list of SqlArgument
  getUdfCandidates(listOfSqlArgument);
}

Reply:
We could for the current `getFunction(final List<SqlType> arguments)` but we'd still need one that accepts `SqlArguments` since the information we need for lambdas is only in `ExpressionTypeManager`, but we can do this approach for `public synchronized KsqlScalarFunction getFunction(final List<SqlType> argTypes)`